.github/workflows/build-and-test.yml

```
@@ -333,7 +333,7 @@ jobs:
       - name: Upload coverage report
         uses: Wandalen/wretry.action@v3.0.1
         with:
-          action: codecov/codecov-action@v3
+          action: codecov/codecov-action@v4
           with: |
             fail_ci_if_error: true
             verbose: true
```

```suggestion
            verbose: true
            token: ${{ secrets.CODECOV_TOKEN }}
```
I think this should enable us to upgrade 👍